### ISSUE TITLE
Update v10 to v11 migration doc 

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#9618]
This issue tracks all the feedback and content we would like to add in for beta 2 in Sprint 18 👀

- [x] Remove notice for `carbon-components`, `carbon-components-react`, and `carbon-icons` having to stay installed
  - [Link](https://v11.carbondesignsystem.com/migrating/guide/develop#how-to-migrate) (inline banner under step 2)
- [x] Add examples for common usage situations with `carbon-components` and what they look like with `@carbon/react`, including ([migration docs](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/11.x-carbon-components.md)):
  - [x] Setting a theme
  - [x] Bringing in a component mixin
  - [x] Using design tokens from the library
- [ ] Add example for using the [`compat`](https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#compatibility) theme through `@carbon/react` (blocked on until this is released: https://github.com/carbon-design-system/carbon/pull/9677)
- [x] Link out to node-sass to sass guide once Taylor publishes it
- [ ] Show example of using the `Layer` component instead of the `light` prop for components that use the light prop ([Link](https://carbon-react-explorer.vercel.app/props/light)) (blocked until this fix is released: https://github.com/carbon-design-system/carbon/pull/9697)
- [ ] (Nice to have) How to use existing [codemods](https://github.com/carbon-design-system/carbon/tree/main/codemods) to update icons or `size` prop
- [ ] Document changes to props that have landed in beta 2 (blocked on that workstream)
